redash/query_runner/elasticsearch.py

```diff
@@ -402,25 +411,34 @@ def name(cls):
     def run_query(self, query, user):
         try:
             error = None
+            index_name = None
+            result_fields = None
 
             logger.debug(query)
-            query_dict = json.loads(query)
-
-            index_name = query_dict.pop("index", "")
-            result_fields = query_dict.pop("result_fields", None)
 
             if not self.server_url:
                 error = "Missing configuration key 'server'"
                 return None, error
 
-            url = "{0}/{1}/_search".format(self.server_url, index_name)
-            mapping_url = "{0}/{1}/_mapping".format(self.server_url, index_name)
+            try:
+                query_dict = json.loads(query)
+                index_name = query_dict.pop("index", "")
+                result_fields = query_dict.pop("result_fields", None)
+                url = "{0}/{1}/_search".format(self.server_url, index_name)
+                params = {"source": json.dumps(query_dict), "source_content_type": "application/json"}
+            except JSONDecodeError:
```

Comment:
Relaying on `JSONDecodeError` to detect SQL queries, means that users who just have some issue with their JSON will get an unexplained error when the fallback behavior is triggered.

How about we create a dedicate query runner for ES SQL?

Reply:
I see your point. My code may change normal behavior of the query runner. Thanks @arikfr 

If I understand correctly, you prefer creating new subclass for ES SQL right?

Reply:
BTW. "ES with SQL plugin" and "Standard ES SQL" (version >=6.3) interfaces are different. For example:

**ES with SQL plugin**
```bash
GET /_sql?sql=SELECT * FROM table LIMIT 5
```

**Standard ES SQL**
```bash
POST /_xpack/sql?format=txt
{
    "query": "SELECT * FROM table LIMIT 5"
}
```

If we implement query runner for ES SQL plugin separately (separate into the new data source), it will be the problem because user may use both normal JSON query and SQL.

Any suggestions?

Reply:
> If I understand correctly, you prefer creating new subclass for ES SQL right?

Yes.

As for the second question, is there a simple way to check what version of ES we're querying? Or maybe, just add a checkbox configuration option to set whether we're using the plugin or not?

Reply:
Good idea! I like the concept that we're having the checkbox to select between JSON query and SQL query. 

Will take care of this.